### PR TITLE
feat: Virtual deployment status & history artifacts

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Added Deployment statuses for virtual env & Artifact in deployment history

Fixes [#AB3097](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=3097)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
By creating & deploying app on virtual cluster & observing deployment status & downloading package from artifacts

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/cicdHistory/types.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ export interface History {
     blobStorageEnabled?: boolean
     isArtifactUploaded?: boolean
     userApprovalMetadata?: UserApprovalMetadataType
+    IsVirtualEnvironment?: boolean
```
Comment (vivek-devtron) [inline src/components/app/details/cicdHistory/types.tsx:32]: IsVirtualEnvironment should be isVirtualEnvironment

File: src/components/app/details/appDetails/utils.tsx [hunk 2]
```diff
@@ -386,6 +386,7 @@ export function addQueryParamToGrafanaURL(url: string, appId: string | number, e
 }
 
 export const processDeploymentStatusDetailsData = (data?: DeploymentStatusDetailsType): DeploymentStatusDetailsBreakdownDataType => {
+
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/utils.tsx:389]: remove unnecessary change

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 3]
```diff
@@ -474,17 +479,29 @@ const HistoryLogs: React.FC<{
...
-                        )}
+                        {triggerDetails.stage !== 'DEPLOY' ||
+                            (triggerDetails.IsVirtualEnvironment && (
+                                <Route path={`${path}/artifacts`}>
+                                    {triggerDetails.IsVirtualEnvironment ? (
+                                        <VirtualHistoryArtifact
+                                            titleName={triggerDetails.artifact}
+                                            appId={appId}
+                                            envId={envId}
+                                            cd_workflow_id={pipelineId}
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/CDDetails.tsx:490]: cd_workflow_id should be cdWorkflowId

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 4]
```diff
@@ -493,7 +527,17 @@ function ChartDeploymentHistory({
                         </div>
                     </div>
                 )}
-                {(selectedDeploymentTabName === DEPLOYMENT_HISTORY_TAB.HELM_GENERATED_MANIFEST || selectedDeploymentTabName === DEPLOYMENT_HISTORY_TAB.VALUES_YAML ) && renderCodeEditor()}
+                {(selectedDeploymentTabName === DEPLOYMENT_HISTORY_TAB.HELM_GENERATED_MANIFEST ||
+                    selectedDeploymentTabName === DEPLOYMENT_HISTORY_TAB.VALUES_YAML) &&
+                    renderCodeEditor()}
+                {selectedDeploymentTabName === DEPLOYMENT_HISTORY_TAB.ARTIFACTS && (
+                    <VirtualHistoryArtifact
+                        titleName={'titleNmae'}
```
Comment (vivek-devtron) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:535]: fix the titleNmae to titleName. Will it be hardcoded value? if yes, we can directly write it as titleName="titleName"

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 5]
```diff
@@ -493,7 +527,17 @@ function ChartDeploymentHistory({
...
+                        appId={appId}
+                        envId={params.envId}
+                        cd_workflow_id={installedAppInfo.installedAppVersionId}
```
Comment (vivek-devtron) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:538]: cd_workflow_id should be cdWorkflowId

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 6]
```diff
@@ -433,7 +444,11 @@ export const Details: React.FC<DetailsType> = ({
         toggleDetailedStatus(false)
     }
 
-    if (!loadingResourceTree && (!appDetails?.resourceTree || appDetails?.resourceTree?.nodes?.length <= 0) && !isVirtualEnvironment) {
+    if (
+        !loadingResourceTree &&
+        (!appDetails?.resourceTree || appDetails?.resourceTree?.nodes?.length <= 0) &&
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/AppDetails.tsx:449]: we can write this (!appDetails?.resourceTree || !appDetails.resourceTree.nodes?.length) &&

File: src/components/app/details/cicdHistory/types.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/app/details/cicdHistory/types.tsx:32]: field is also capatalise, will do with BE changes

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 4 above)
Comment (milindk8) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: we are giving dynamic value now

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 5 above)
Comment (milindk8) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: changed

**Review by vivek-devtron — APPROVED** [review]
